9/src/part1.c

```diff
@@ -0,0 +1,85 @@
+// Link: https://adventofcode.com/2022/day/9
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+#define MATRIX_SIZE 500
+
+typedef struct {
+    int x, y;
+} point;
+
+bool visited[MATRIX_SIZE][MATRIX_SIZE] = {0};
+
+void move_tail(point *head, point *tail) {
+    // If head and tail are on the same line:
+    if (head->x == tail->x) { // vertical line
+        if (abs(head->y - tail->y) <= 1)return;
+        if (head->y < tail->y) tail->y--;
+        else if (head->y > tail->y) tail->y++;
```

Comment:
```suggestion
        if (abs(head->y - tail->y) > 1) {
            tail->y += (head->y > tail->y) ? 1 : -1;
        }
```
perform this optimization for rest cases